packages/instant/webpack.config.js

```diff
@@ -11,22 +12,104 @@ const GIT_SHA = childProcess
+    if (env && env.dogfood) {
+        return 'dogfood';
+    } else if (env && env.staging) {
+        return 'staging';
     }
 
-    if (modeName === 'development') {
-        return process.env[HEAP_DEVELOPMENT_ENV_VAR_NAME];
+    // argv.mode should be 'development' or 'production'
+    return argv.mode;
+};
+
+const getHeapAnalyticsId = environmentName => {
+    if (environmentName === 'production') {
+        return process.env['INSTANT_HEAP_ANALYTICS_ID_PRODUCTION'];
+    }
+
+    if (environmentName === 'development' || environmentName === 'dogfood' || environmentName === 'staging') {
+        return process.env['INSTANT_HEAP_ANALYTICS_ID_DEVELOPMENT'];
     }
 
     return undefined;
 };
 
+const ROLLBAR_PUBLISH_TOKEN_ENV_NAME = 'INSTANT_ROLLBAR_PUBLISH_TOKEN';
+const ROLLBAR_CLIENT_TOKEN_ENV_NAME = 'INSTANT_ROLLBAR_CLIENT_TOKEN';
+const getRollbarPlugin = environmentName => {
+    if (!environmentName) {
+        return undefined;
+    }
+
+    const publishToken = process.env[ROLLBAR_PUBLISH_TOKEN_ENV_NAME];
+    if (!publishToken) {
+        return undefined;
+    }
+
+    let rollbarPublicPath;
+    if (environmentName === 'dogfood') {
+        rollbarPublicPath = 'http://0x-instant-dogfood.s3-website-us-east-1.amazonaws.com';
+    } else if (environmentName === 'staging') {
+        rollbarPublicPath = 'http://0x-instant-staging.s3-website-us-east-1.amazonaws.com';
+    } // TODO(sk): When we decide on JS cdn, add public path here
+
+    if (!rollbarPublicPath) {
+        console.log('No rollbar public path');
+        return undefined;
+    }
+
+    const rollbarPluginOptions = {
+        accessToken: publishToken,
+        version: GIT_SHA,
+        publicPath: rollbarPublicPath,
+    };
+    return new RollbarSourceMapPlugin(rollbarPluginOptions);
+};
+const validateRollbarPresence = (environmentName, rollbarEnabled, rollbarSourceMapPlugin) => {
+    const requiresRollbar = environmentName === 'dogfood' || environmentName === 'staging';
+    if (!requiresRollbar) {
+        return;
+    }
+    if (!rollbarEnabled || !rollbarSourceMapPlugin) {
+        throw new Error(
+            `Rollbar env vars must be set to build for ${environmentName}. Please set ${ROLLBAR_CLIENT_TOKEN_ENV_NAME} to a rollbar access token with post_client_item permissions, and ${ROLLBAR_PUBLISH_TOKEN_ENV_NAME} to a rollbar access token with post_server_item permissions.`,
+        );
+    }
+};
+
 module.exports = (env, argv) => {
+    const environmentName = getEnvironmentName(env, argv);
     const outputPath = process.env.WEBPACK_OUTPUT_PATH || 'umd';
+
+    const envVars = {
+        GIT_SHA: JSON.stringify(GIT_SHA),
+        NPM_PACKAGE_VERSION: JSON.stringify(process.env.npm_package_version),
+        HEAP_ANALYTICS_ID: getHeapAnalyticsId(environmentName),
+        INSTANT_ENVIRONMENT: JSON.stringify(environmentName),
+        ROLLBAR_CLIENT_TOKEN: JSON.stringify(process.env[ROLLBAR_CLIENT_TOKEN_ENV_NAME]),
+    };
+
+    const canRollbarBeEnabled =
```

Comment:
The idea behind `INSTANT_ROLLBAR_FORCE_DEVELOPMENT_REPORT` is guarding against having  `INSTANT_ROLLBAR_CLIENT_TOKEN` set in your environment and accidentally reporting in the development environment